pkg/container-collection/options.go

```diff
@@ -432,3 +438,280 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
 		return nil
 	}
 }
+
+// WithKubernetesOwnerRefEnrichment automatically adds owner references metadata
+func WithKubernetesOwnerRefEnrichment(nodeName string) ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			return fmt.Errorf("couldn't get Kubernetes config: %s", err)
+		}
+		dynamicClient, err := dynamic.NewForConfig(config)
+		if err != nil {
+			return fmt.Errorf("couldn't start dynamic Kubernetes client: %s", err)
+		}
+
+		type ResourceParameters struct {
+			groupVersion schema.GroupVersionResource
+			namespace    string
+			listOpt      metav1.ListOptions
+		}
+
+		getResourceList := func(dynamic dynamic.Interface, params *ResourceParameters) (
```

Comment:
I think defining these as normal functions instead of lambda ones will improve code readability

Reply:
Now there is only one function. I think it ok now 🙂 .